### ISSUE TITLE
cabal-install-1.16.0.2 depends on HTTP-4000.2.6 which failed to install

### ISSUE THREAD

**Issue by igouy** [haskell/cabal#1137]
Linux x64 3.2.0-33-generic #52-Ubuntu SMP Thu Oct 18 16:29:15 UTC 2012 x86_64 x86_64 x86_64 GNU/Linux

$ cabal install cabal-install
...
...
In-place registering Cabal-1.16.0...
Installing library in /home/dunham/.cabal/lib/Cabal-1.16.0/ghc-7.6.1
Registering Cabal-1.16.0...
cabal: Error: some packages failed to install:
HTTP-4000.2.6 failed during the building phase. The exception was:
ExitFailure 1
cabal-install-1.16.0.2 depends on HTTP-4000.2.6 which failed to install.

$ cabal install HTTP-4000.2.6
...
...
Configuring HTTP-4000.2.6...
Preprocessing library HTTP-4000.2.6...
Preprocessing test suites for HTTP-4000.2.6...
Building HTTP-4000.2.6...
[ 1 of 18] Compiling Network.HTTP.Base64 ( Network/HTTP/Base64.hs, dist/build/Network/HTTP/Base64.o )
[ 2 of 18] Compiling Network.HTTP.MD5Aux ( Network/HTTP/MD5Aux.hs, dist/build/Network/HTTP/MD5Aux.o )
[ 3 of 18] Compiling Paths_HTTP       ( dist/build/autogen/Paths_HTTP.hs, dist/build/Paths_HTTP.o )

dist/build/autogen/Paths_HTTP.hs:21:13: Not in scope: `catch'

dist/build/autogen/Paths_HTTP.hs:22:13: Not in scope: `catch'

dist/build/autogen/Paths_HTTP.hs:23:14: Not in scope: `catch'

dist/build/autogen/Paths_HTTP.hs:24:17: Not in scope: `catch'
cabal: Error: some packages failed to install:
HTTP-4000.2.6 failed during the building phase. The exception was:
ExitFailure 1
